src/GitVersionCore.Tests/StringFormatWithExtensionTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+
+using GitVersion;
+using NUnit.Framework;
+
+namespace GitVersionCore.Tests
+{
+    [TestFixture]
+
+    public class StringFormatWithExtensionTests
+    {
+        [Test]
+        public void FormatWith_NoTokens()
+        {
+            var propertyObject = new { };
+            var target = "Some String without tokens";
+            var expected = target;
+            var actual = target.FormatWith(propertyObject);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void FormatWith_SingleSimpleToken()
+        {
+            var propertyObject = new { SomeProperty = "SomeValue" };
+            var target = "{SomeProperty}";
+            var expected = "SomeValue";
+            var actual = target.FormatWith(propertyObject);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void FormatWith_MultipleTokensAndVerbatimText()
+        {
+            var propertyObject = new { SomeProperty = "SomeValue", AnotherProperty = "Other Value" };
+            var target = "{SomeProperty} some text {AnotherProperty}";
+            var expected = "SomeValue some text Other Value";
+            var actual = target.FormatWith(propertyObject);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void FormatWith_EnvVarToken()
+        {
+            Environment.SetEnvironmentVariable("GIT_VERSION_TEST_VAR", "Env Var Value");
+            var propertyObject = new { };
+            var target = "{env:GIT_VERSION_TEST_VAR}";
+            var expected = "Env Var Value";
+            var actual = target.FormatWith(propertyObject);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void FormatWith_EnvVarTokenWithFallback()
+        {
+            Environment.SetEnvironmentVariable("GIT_VERSION_TEST_VAR", "Env Var Value");
+            var propertyObject = new { };
+            var target = "{env:GIT_VERSION_TEST_VAR??fallback}";
```

Comment:
Are spaces allowed around the null propagation operator? That would be a readability improvement, imho. Is that hard to add?

Reply:
Addressed in the latest patch.